packages/engine/lib/error/src/frame.rs

```diff
@@ -33,10 +40,27 @@ use crate::{
 pub struct Frame {
     inner: ManuallyDrop<Box<FrameRepr>>,
     location: &'static Location<'static>,
-    source: Option<Box<Frame>>,
+    pub(crate) source: Option<Box<Frame>>,
```

Comment:
Required for `Frames` implementation

Reply:
Fair call
```suggestion
    /// Required for [`Frames`] implementation for non-nightly builds
    ///
    /// [`Frames`]: crate::iter::Frames
    pub(crate) source: Option<Box<Frame>>,
```

Reply:
Wait is this for the `source` field or the reasoning for `pub(crate)`?

Reply:
We could do this on a few other places nearby in the code if you'd like

Reply:
I haven't found any code where this would be needed. This is the only location, where a field needs to be crate-visible, for functions this is only really helpful for `generalized` which already has reasoning on it.